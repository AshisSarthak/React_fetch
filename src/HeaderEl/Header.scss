.App-header {
  background-color: #EEE;
  height: 40px;
  display: flex;
  align-items: center;
  justify-content: space-evenly;
  font-size: calc(10px + 2vmin);
  color: #666;
  border-bottom: 1px solid #eee;
}
.backBtn {
  display: flex;
  font-size: 16px;
  align-items: center;
  cursor: pointer;

  & .leftArrow{
    background-image: url('../resources/left-arrow.svg');
    background-repeat: no-repeat;
    background-size: 20px;
    height: 20px;
    width: 20px;
    margin-right: 5px;
    cursor: pointer;
    &:hover{
      text-decoration: underline;
    }
  }
  

  &:hover{
    text-decoration: underline;
  }
}
.user-as {
  width: 300px;
  height: 30px;
  border: 1px solid #ccc;
  border-radius: 20px;
  text-indent: 20px;
  font-size: 15px;
  background-image: url('../resources/search.png');
  background-repeat: no-repeat;
  background-size: 25px;
  background-position: 96% 3px;
  outline: none;
}