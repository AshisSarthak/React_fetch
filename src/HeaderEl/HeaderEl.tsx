import React, { ChangeEvent } from 'react';
import './Header.scss';

interface HeaderProps {
  showBackButton: boolean;
  handleBackClick: () => void;
  handleUserSearch: (userName: string) => void;
}
const HeaderEl = (props: HeaderProps) => {
  const searchUser=($event: ChangeEvent<HTMLInputElement>)=>  {
    props.handleUserSearch($event.currentTarget.value);
  }
  return (
      <header className="App-header">
        {
          props.showBackButton && (<div className="backBtn" onClick={props.handleBackClick}>
          <div className="leftArrow"></div>
            Back
          </div>)
        }
        <div className="seach-user">
          <input className='user-as' onChange={(e: ChangeEvent<HTMLInputElement>) => searchUser(e)} placeholder="Search User"/>
        </div>
      </header>
  );
}

export default HeaderEl;
