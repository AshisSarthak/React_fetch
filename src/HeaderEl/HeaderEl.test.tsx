import React from "react";
import { shallow } from "enzyme";
import HeaderEl from "./HeaderEl";

describe("HeaderEl Componnet", () => {
  it("default render with no back arrow", () => {
    const backClick = jest.fn();
    const userSearch = jest.fn();
    const wrapper = shallow(
      <HeaderEl
        showBackButton={false}
        handleBackClick={backClick}
        handleUserSearch={userSearch}
      />
    );
    expect(wrapper).toMatchSnapshot();

    const backArrow = wrapper.find("div.leftArrow");
    expect(backArrow.length).toBe(0);
  });

  it("default render with back arrow", () => {
    const backClick = jest.fn();
    const userSearch = jest.fn();
    const wrapper = shallow(
      <HeaderEl
        showBackButton={true}
        handleBackClick={backClick}
        handleUserSearch={userSearch}
      />
    );
    expect(wrapper).toMatchSnapshot();

    const backArrow = wrapper.find("div.leftArrow");
    expect(backArrow.length).toBe(1);

    const backBtn = wrapper.find("div.backBtn").at(0);
    backBtn.simulate("click");
    expect(backClick).toBeCalled();

    const inputEL = wrapper.find("input").at(0);
    inputEL.simulate("change", { currentTarget: { value: "newval" } });
    expect(userSearch).toBeCalledWith("newval");
  });
});
