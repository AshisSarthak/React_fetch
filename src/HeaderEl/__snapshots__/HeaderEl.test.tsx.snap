// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderEl Componnet default render with back arrow 1`] = `
<header
  className="App-header"
>
  <div
    className="backBtn"
    onClick={[MockFunction]}
  >
    <div
      className="leftArrow"
    />
    Back
  </div>
  <div
    className="seach-user"
  >
    <input
      className="user-as"
      onChange={[Function]}
      placeholder="Search User"
    />
  </div>
</header>
`;

exports[`HeaderEl Componnet default render with no back arrow 1`] = `
<header
  className="App-header"
>
  <div
    className="seach-user"
  >
    <input
      className="user-as"
      onChange={[Function]}
      placeholder="Search User"
    />
  </div>
</header>
`;
